smqtk_classifier/interfaces/classify_image.py

```diff
@@ -0,0 +1,145 @@
+import abc
+import logging
+from typing import Generator, Hashable, Iterable, Iterator, Sequence, Union
+
+import numpy as np
+from PIL import Image
+
+from smqtk_core import Plugfigurable
+
+from smqtk_classifier.interfaces.classification_element import (
+    CLASSIFICATION_DICT_T
+)
+
+
+IMAGE_ITER_T = Union[Image.Image, Iterable[Image.Image]]
+LOG = logging.getLogger(__name__)
+
+
+class ClassifyImage (Plugfigurable):
+    """
+    Interface for algorithms that classify input images into discrete
+    labels and/or label confidences.
+    """
+
+    @abc.abstractmethod
+    def get_labels(self) -> Sequence[Hashable]:
+        """
+        Get the sequence of class labels that this classifier can classify
+        images into. This includes the negative or background label if the
+        classifier embodies such a concept.
+
+        :return: Sequence of possible classifier labels.
+        :rtype: collections.abc.Sequence[collections.abc.Hashable]
+
+        :raises RuntimeError: No model loaded.
+
+        """
+
+    @abc.abstractmethod
+    def _classify_arrays(self, img_iter: IMAGE_ITER_T) -> Iterator[CLASSIFICATION_DICT_T]:
+        """
+        Overridable method for classifying an iterable of PIL.Image.Image
+        whose vectors should be classified.
```

Comment:
Don't need to mention "vectors" here (assuming a left-over from the descriptor variant)

Reply:
Some other bits of this doc-string also seem like copy-paste left-overs.